### PR TITLE
Langauge Server Progress Notifications

### LINKED ISSUE #1572: UX: Index progress notifications intrusive
The notifications when indexing a large project are quite intrusive and annoying. It also slows down and can lock up VS Code.

image

### PR DESCRIPTION
**Pull request by dantleech**
Fixes #1572 

replaces #1792

### CODE REVIEW

**Comment by dantleech** [comment]
cc @innocenzi

File: lib/Extension/LanguageServer/Handler/DebugHandler.php [hunk 1]
```diff
@@ -137,6 +142,7 @@ public function status(): Promise
             '  documents: ' . $this->workspace->count(),
             '  services: ' . (string)json_encode($this->serviceManager->runningServices()),
             '  diagnostics: ' . (string)$this->diagnosticProvider->name(),
+            '  watcher: ' . $this->watcher->describe(),
```
Comment (dantleech) [inline lib/Extension/LanguageServer/Handler/DebugHandler.php:145]: the only "useful" thing about the welcome message was showing the watcher status, but I'm not sure if anybody really cared much about it. I've added it to the status report instead.

File: lib/Extension/LanguageServerIndexer/Handler/IndexerHandler.php [hunk 2]
```diff
@@ -35,10 +35,6 @@ class IndexerHandler implements Handler, ServiceProvider
 
     private ClientApi $clientApi;
 
-    private ServiceManager $serviceManager;
-
-    private bool $doReindex = false;
```
Comment (dantleech) [inline lib/Extension/LanguageServerIndexer/Handler/IndexerHandler.php:40]: not used

File: lib/Extension/LanguageServerIndexer/Handler/IndexerHandler.php [hunk 3]
```diff
@@ -92,13 +91,17 @@ public function indexer(CancellationToken $cancel): Promise
...
-                        'Indexed %s/%s (%s%%) %s',
-                        $index,
-                        $size,
-                        number_format($index / $size * 100, 2),
-                        $usage->memoryUsageFormatted()
-                    ));
+                    $this->clientApi->workDoneProgress()->report(
+                        $token,
+                        sprintf(
+                            '%s/%s (%s%%) %s',
```
Comment (innocenzi) [inline lib/Extension/LanguageServerIndexer/Handler/IndexerHandler.php:97]: Just a little nitpick:
```suggestion
                            '%s/%s (%s%%, %s)',
```

**Pull request merged**